pkg/machines/components/vmSnapshotsRevertModal.jsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal } from 'patternfly-react';
+import { Button } from '@patternfly/react-core';
+
+import cockpit from 'cockpit';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { revertSnapshot } from '../libvirt-dbus.js';
+
+const _ = cockpit.gettext;
+
+export class RevertSnapshotModal extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            dialogError: undefined,
+            inProgress: false,
+        };
+
+        this.revert = this.revert.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+    }
+
+    revert() {
+        this.setState({ inProgress: true });
+        const { vm, snap } = this.props;
+
+        revertSnapshot({ connectionName: vm.connectionName, domainPath: vm.id, snapshotName: snap.name })
+                .then(this.props.onClose, exc => {
+                    this.setState({ inProgress: false });
+                    this.dialogErrorSet(_("Could not revert to snapshot"), exc.message);
+                });
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    render() {
+        const { idPrefix, snap, onClose } = this.props;
+
+        return (
+            <Modal id={`${idPrefix}-snapshot-${snap.name}-modal`} show onHide={onClose}>
+                <Modal.Header>
+                    <Modal.CloseButton onClick={onClose} />
+                    <Modal.Title>{ cockpit.format(_("Revert to Snapshot $0"), snap.name) }</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    { cockpit.format(_("Confirm this action")) }
```

Comment:
Needs some language around how this is a possibly destructive action.

Reply:
Same for the delete dialog. But I'm too shit at reading code, because I can't find that...

Reply:
Yes. As I mentioned in my response to garrett's comment, we could put some warning saying something like you will loose all "non-snapshotted" progress when you revert, or/plus perhaps even allow user to create some sort of "temporary" snapshot before reverting so they do not loose all the progress.

Reply:
Maybe something like: _"Reverting to this snapshot will take the VM back to the time of the snapshot and the current state will be lost, along with any data not snapshotted"_

Might be a bit long, so suggestions for tweaks welcome!

Reply:
_plus perhaps even allow user to create some sort of "temporary" snapshot before reverting so they do not loose all the progress._

Not a bad idea, but let's do that in another PR in that case.

Reply:
> Maybe something like: _"Reverting to this snapshot will take the VM back to the time of the snapshot and the current state will be lost, along with any data not snapshotted"_

"snapshotted" looks like a weird word, perhaps "captured in a snapshot" ?

Basically by loosing state and data we only mean loosing data on disk and in memory (that's the only thing snapshots capture). And in some snapshot variations (some are only to be implemented by libvirt), some snapshots only capture VM's memory, and some only capture VM's disks state.. Changes to external things like Network interfaces or changing number of CPUs are not captured in snapshots. 

So I cannot thing of sentence which says that disk data might be lost ONLY if they are captured and thus overwritten in the snapshot, same with memory and keep it short